py4j-java/src/main/java/py4j/reflection/TypeConverter.java

```diff
@@ -94,8 +98,36 @@ public Object convert(Object obj) {
 		return newObject;
 	}
 
+	/**
+	 * Convert remaining varags into an array
+	 * @param arrayClass 
+	 * @param index
+	 * @param newArguments
+	 * @return converted objects
+	 */
+	public Object[] convert(Class<?> arrayClass, int index, Object[] newArguments) {
+		Object lastArg = Array.newInstance(arrayClass, newArguments.length - index);
+		int count = 0;
+		for (int i = index; i < newArguments.length; i++) {
+			Array.set(lastArg, count++, newArguments[i]); // TODO convert?
```

Comment:
The problem is that we do not convert variable arrays of, say, doubles to floats. Or, arrays of non-primitives.